tests/subyt/test_unique_filter_source.py

```
@@ -0,0 +1,148 @@
+    with upfs as items:
+        uniq_lead9_orcids = [item for item in items]
+        assert len(uniq_lead9_orcids) == 3
+        print(f"{uniq_lead9_orcids=}")
+        for i in range(0, 3):
+            assert uniq_lead9_orcids[i]["orcid"].startswith(f"0000-000{i + 1}")
+            assert uniq_lead9_orcids[i]["name"] == expected_names[i]
+
+
+def test_valid_unique_pattern() -> None:
+    fdls = FixedDataListSource()
+    with pytest.raises(ValueError):
```

Ruff (0.8.2)

`pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)